Source/WebKit/UIProcess/API/glib/WebKitDownload.cpp

```diff
@@ -126,9 +135,23 @@ static void webkitDownloadGetProperty(GObject* object, guint propId, GValue* val
     }
 }
 
+static void maybeFinishDecideDestination(WebKitDownload* download)
+{
+    if (download->priv->decideDestinationCallback) {
+#if ENABLE(2022_GLIB_API)
+        const char* destination = download->priv->destination.data();
+#else
+        GUniquePtr<char> destinationPath(g_filename_from_uri(download->priv->destination.data(), nullptr, nullptr));
+        const char* destination = destinationPath.get();
+#endif
+        (*download->priv->decideDestinationCallback)(download->priv->allowOverwrite ? AllowOverwrite::Yes : AllowOverwrite::No, destination ? String::fromUTF8(destination) : emptyString());
```

Comment:
String::fromUTF8 will return an empty string if destination is nullptr, I think. But this is not what we want, we say we use the suggested filename to sabe it under the downloads dir when not handled. If this was handled, then we should have a destination.

Reply:
Here we don't know whether it was handled or not. This code runs in both conditions.